marklogic-data-hub/src/main/resources/ml-modules/root/com.marklogic.smart-mastering/survivorship/merging/base.xqy

```diff
@@ -352,16 +350,46 @@ declare function merge-impl:rollback-merge(
     else
       $all-contributing-uris[(@last-merge|../last-merge) = $last-merge-dateTime] ! fn:string(.)
   let $merge-doc-in-previous := $previous-uris = $merged-doc-uri
+  let $_trace := if (xdmp:trace-enabled($const:TRACE-MERGE-RESULTS)) then
+          let $doc-prefix := "cts.doc('" || $merged-doc-uri || "')"
+          return (
+            xdmp:trace($const:TRACE-MERGE-RESULTS, $doc-prefix || " previous URIs: " || xdmp:to-json-string($previous-uris)),
+            xdmp:trace($const:TRACE-MERGE-RESULTS, $doc-prefix || " last merge dateTime: " || xdmp:to-json-string($last-merge-dateTime)),
+            xdmp:trace($const:TRACE-MERGE-RESULTS, $doc-prefix || " latest auditing receipt: " || xdmp:to-json-string($latest-auditing-receipt-for-doc))
+          )
+        else
+          ()
   where fn:exists(($latest-auditing-receipt-for-doc,$last-merge-dateTime))
   return (
+    let $remerged-doc :=
```

Comment:
This seems like a lot of significant changes based on failures in QS e2e tests; the fact that no tests were failing in the ./marklogic-data-hub is a real concern, because we don't want to be relying on QS tests that are poorly understood (I'm looking at ./web/e2e and can't figure out what's going on - the ./qa-data folder makes this look like a DHF 4 project). 

What are the actual problems surfaced by this JIRA ticket? The main test fix I see is that "targetEntity" is now correctly changed to "targetEntityType", but it looks like we have a lot more logic being implemented here, and I can't tell what problem is being solved by looking at the tests and at the JIRA ticket.